tests/h/services/search_index/_queue_test.py

```diff
@@ -36,30 +38,48 @@ def test_it(self, db_session, factories, queue, now):
                     tag="test_tag",
                     priority=1,
                     kwargs={
-                        "annotation_id": URLSafeUUID.url_safe_to_hex(annotation.id)
+                        "annotation_id": URLSafeUUID.url_safe_to_hex(annotation.id),
+                        "force": False,
                     },
                 )
             ),
         ]
 
 
 class TestAddAnnotationsBetweenTimes:
-    def test_it(self, annotation_ids, db_session, queue):
+    @pytest.mark.parametrize("force", [True, False])
+    def test_it(self, annotation_ids, db_session, queue, force):
         queue.add_annotations_between_times(
             datetime_.datetime(2020, 9, 9),
             datetime_.datetime(2020, 9, 11),
             "test_tag",
+            force,
         )
 
-        annotation_ids_added_to_jobs_table = [
-            URLSafeUUID.hex_to_url_safe(job.kwargs["annotation_id"])
-            for job in db_session.query(Job)
-        ]
         assert (
-            annotation_ids_added_to_jobs_table
-            == Any.list.containing(annotation_ids).only()
+            db_session.query(Job).all()
+            == Any.list.containing(
+                [
+                    Any.instance_of(Job).with_attrs(
+                        {
+                            "tag": "test_tag",
+                            "name": "sync_annotation",
+                            "scheduled_at": Any.instance_of(datetime_.datetime),
+                            "priority": 1000,
+                            "kwargs": {
+                                "annotation_id": str(
+                                    uuid.UUID(
+                                        URLSafeUUID.url_safe_to_hex(annotation_id)
+                                    )
+                                ),
+                                "force": force,
+                            },
+                        }
+                    )
+                    for annotation_id in annotation_ids
+                ]
+            ).only()
```

Comment:
Giving h-matchers a workout here. `URLSafeUUID.url_safe_to_hex()` actually doesn't return strings that match those in the DB. There's further processing that happens after that because the DB is using a UUID type. So you have to do `str(uuid.UUID(URLSafeUUID.url_safe_to_hex(annotation_id)))`

Reply:
Oh good gravy.

These comparisons get pretty long and rambling, but they still read ok mostly.